src/servers/grpc_server_v2.cc

```diff
@@ -319,7 +319,12 @@ class HandlerState {
     // The step of the entire context.
     Steps step_;
 
-    // True if this context should finish with OK status, false if
+    // The status of the requests in the stream. If OK then all the requests
+    // in the stream were successfull, otherwise stores the status of the
+    // first failing request in the stream.
+    grpc::Status request_status_;
```

Comment:
Why do we need this? Why does the overall status handling for the stream have to change from what we have now? Even if individual requests in the stream fail that doesn't necessarily impact the overall stream status.